packages/interface/src/lib.ts

```diff
@@ -397,12 +397,27 @@ export interface PrincipalParser {
  * Library also provides utility functions for combining multiple
  * SignerImporters into one.
  */
-export interface SignerImporter<Self extends Signer = Signer> {
-  from(archive: SignerArchive<Self>): Self
+export interface SignerImporter<
+  ID extends DID = DID,
+  Code extends number = number
+> {
+  from(archive: SignerArchive<ID, Code>): Signer<ID, Code>
 }
 
-export interface Signer<M extends string = string, A extends number = number>
-  extends UCANSigner<M, A> {
+export interface Signer<ID extends DID = DID, A extends number = number>
+  extends Principal<ID>,
+    Verifier<ID, A> {
+  signer: Signer<ID, A>
+  verifier: Verifier<ID, A>
+
+  signatureAlgorithm: string
```

Comment:
It is serves both as humane version of varsig code and in case of non standard sig algorithms it’s encoded into varsig itself so it could be used in jwt for sig algorithm type field

Reply:
got it. So if there were a comment on this `string`, it may want to point to this IANA registry (which contains jwt 'alg' fields, e.g. 'EdDSA'): https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms